### ISSUE TITLE
Result_format "COMPLETE" does return "unexpected_index_list": null

### ISSUE THREAD

**Issue by cwerner** [fivetran/great_expectations#3736]
Hi. I try to run a validation on a PostgreSQL table that I also process with a query statement and want to obtain the index of rows that fail the expectation. 

I'm using a `RuntimeBatchRequest` and want to inspected the result object via `unexpected_index_list` and then flag the row in the database with a QC label. I'm not sure what's going on?

Steps to reproduce the behaviour:

This is my code:
```python
import great_expectations as ge
from great_expectations.cli.datasource import sanitize_yaml_and_save_datasource, check_if_datasource_name_exists
from great_expectations.core.batch import RuntimeBatchRequest

context = ge.get_context()

# specify the username, password etc ... (omitted)

example_yaml = f"""
name: {datasource_name}
class_name: Datasource
execution_engine:
  class_name: SqlAlchemyExecutionEngine
  credentials:
    host: {host}
    port: '{port}'
    username: {username}
    password: {password}
    database: {database}
    drivername: postgresql
data_connectors:
  default_runtime_data_connector_name:
    class_name: RuntimeDataConnector
    batch_identifiers:
      - default_identifier_name
  default_inferred_data_connector_name:
    class_name: InferredAssetSqlDataConnector
    name: whole_table"""

context.test_yaml_config(yaml_config=example_yaml)
sanitize_yaml_and_save_datasource(context, example_yaml, overwrite_existing=True)


var = 'airtemp_Avg'

statement = f"""
SELECT timestamp, value
FROM sensordata
WHERE variable='{var}' AND timestamp >(
    SELECT MAX(DATE_TRUNC('day', timestamp)) 
    FROM sensordata
) LIMIT 5; 
"""

batch_request = RuntimeBatchRequest(
    datasource_name="timescale",
    data_connector_name="default_runtime_data_connector_name",
    data_asset_name="default_name",  # this can be anything that identifies this data
    runtime_parameters={
        "query": statement,
        },
    batch_identifiers={"default_identifier_name": "default_identifier"},
)

context.create_expectation_suite(
    expectation_suite_name="test_suite_xxx", overwrite_existing=True
)

validator = context.get_validator(
    batch_request=batch_request, expectation_suite_name="test_suite_xxx"
)

validator.head()
```

This gives me the expected data:  

 timestamp | value
 -- | --
2020-01-09 00:01:00 | 2.2
2020-01-09 00:02:00 | 2.2
2020-01-09 00:03:00 | 2.1
2020-01-09 00:04:00 | 2.1
2020-01-09 00:05:00 | 2.0

```python
res = validator.expect_column_values_to_be_between("value", min_value=-35, max_value=2.1, result_format='COMPLETE')
```

This leads to some strange results. Not sure if this is related to my SQL query?!?
```
{
  "success": false,
  "result": {
    "element_count": 5,
    "unexpected_count": 2,
    "unexpected_percent": 40.0,
    "partial_unexpected_list": [
      2.2,
      2.2
    ],
    "missing_count": 0,
    "missing_percent": 0.0,
    "unexpected_percent_total": 40.0,
    "unexpected_percent_nonmissing": 40.0,
    "partial_unexpected_index_list": null,
    "partial_unexpected_counts": [
      {
        "value": 2.2,
        "count": 2
      }
    ],
    "unexpected_list": [
      2.2,
      2.2
    ],
    "unexpected_index_list": null
  },
  "meta": {},
  "exception_info": {
    "raised_exception": false,
    "exception_traceback": null,
    "exception_message": null
  }
}
```

**Expected behavior**
Return the indices where the expectation fails.

**Environment:**
 - Operating System: Linux
 - Great Expectations Version: 0.13.43

Maybe this is related to issue #3225? 

Also I came across this discussion: https://discuss.greatexpectations.io/t/can-great-expectations-separate-sql-database-records-that-fail-tests-from-my-good-data/640 
Might this be the cause that SQL queries currently do not return the index?

**Comment by cwerner** [comment]
I came across this discussion. Might this be the cause that SQL queries currently do not return the index?

https://discuss.greatexpectations.io/t/can-great-expectations-separate-sql-database-records-that-fail-tests-from-my-good-data/640

**Comment by cdkini** [comment]
Hey @cwerner! Thanks for opening up this issue. We'll take this information and review internally. Appreciate your patience!

**Comment by abekfenn** [comment]
@cwerner see https://discuss.greatexpectations.io/t/can-great-expectations-separate-sql-database-records-that-fail-tests-from-my-good-data/640 and https://github.com/great-expectations/great_expectations/issues/3195
TL;DR This is an open feature request but there are workarounds that may work for your use case. In my case, the workaround mentioned would not work as there may be a value for row_condition that is acceptable and unacceptable for another row_condition. Therefore, removing values that are in the unexpected_value_list could result in "expected" data being removed.

Importantly, https://github.com/great-expectations/great_expectations/pull/3346 started to make progress towards this.

**Comment by cdkini** [comment]
Thank you @abekfenn! @cwerner please let me know if the above response provides any clarity here.

**Comment by cwerner** [comment]
Hi @cdkini 

In my case, I think this is simply not implemented for SQL data stores (yet?)... 🤷‍♂️

I currently use a workaround though. 

Basically do the following:
1. `SELECT` on the table and convert to an `pandas dataframe` 
2. Call the validation on this temporary pandas dataframe. 
3. Parse the result object and extract the indices of rows with problems. 
4. Use the indices to get the primary keys for the corresponding entries in the database and set the QC flag there.

Not ideal, but works for now...

**Comment by talagluck** [comment]
Hi @cwerner - yes, this is correct. Currently, `unexpected_index_list` is only implemented for `pandas`. I also wanted to point you to the new `include_unexpected_rows` flag that we have - this will return the entirety of each row that has failed an expectation (see here). This has the potential to return huge amounts of data within your ValidationResults, so I would advise caution when using this.

Finally, we have this feature issue which we hope to implement in the not-too-distant future, which will enable the return of select columns (like IDs or PKs) for rows that fail.

I'm going to close this for now, but thank you again for the question!
